src/manifests/manifest.py

```diff
@@ -51,15 +51,17 @@ def from_path(cls, path):
 
     @classmethod
     def compact(cls, d):
-        result = {}
-        for k, v in d.items():
-            if isinstance(v, dict):
-                nested = cls.compact(v)
-                if nested:
-                    result[k] = nested
-            elif v and v != []:
-                result[k] = v
-        return result
+        if isinstance(d, list):
+            return list(map(lambda i: cls.compact(i), d))
+        elif isinstance(d, dict):
+            result = {}
+            for k, v in d.items():
```

Comment:
Can we be more specific with `k` and `v` here and for the rest below?

Reply:
Well, they are key and value, but nothing specific :)